moped-etl/arcgis/components_to_agol.py

```diff
@@ -65,19 +64,49 @@ def main():
     logger.info(f"{len(data)} component records to process")
 
     # lines and points must be stored in different layers in AGOL
-    all_features = {"lines": [], "points": []}
+    all_features = {"lines": [], "points": [], "combined": []}
 
     for component in data:
-        feature, is_point_feature = make_esri_feature(**component)
+        # extract geometry and line geometry from component data
+        # for line features, the line geometry is redundant/identical to geometry
+        # for point features, it is the buffered ring around the point as defined
+        # in the Moped component view
+        geometry = component.pop("geometry")
+        line_geometry = component.pop("line_geometry")
+
+        if not geometry:
+            continue
+
+        esri_geometry_key = get_esri_geometry_key(geometry)
+        feature = make_esri_feature(
+            esri_geometry_key=esri_geometry_key, geometry=geometry, attributes=component
+        )
+
+        # adds a special `source_geometry_type` column that will be useful on the `combined` layer
+        # so that we can keep track of if the original feature was a point or line geometry
+        feature["attributes"]["source_geometry_type"] = (
+            "point" if esri_geometry_key == "points" else "line"
+        )
+
         if not feature:
             continue
 
-        if is_point_feature:
+        if esri_geometry_key == "points":
             all_features["points"].append(feature)
+            # create the point -> line feature
+            if line_geometry["type"] == "LineString":
+                # if we're converting a single point to a line, we need to convert that line geom to
+                # a multi-line geometry
+                line_geometry["coordinates"] = [line_geometry["coordinates"]]
+            line_feature = make_esri_feature(
+                esri_geometry_key="paths", geometry=line_geometry, attributes=component
+            )
+            all_features["combined"].append(line_feature)
```

Comment:
this looks like more refactoring than it really was bc i tweaked some awkward function signatures. tldr: we use the `line_geomtery` column to create a third group of features that is the superset of points + lines.